### ISSUE TITLE
Compilation error on OSX (sendfile related)

### ISSUE THREAD

**Issue by saghul** [joyent/libuv#568]
```
src/unix/fs.c: In function ‘uv__fs_sendfile’:
src/unix/fs.c:259: warning: passing argument 4 of ‘sendfile’ makes pointer from integer without a cast
src/unix/fs.c:259: warning: passing argument 6 of ‘sendfile’ makes integer from pointer without a cast
src/unix/fs.c:259: error: too many arguments to function ‘sendfile’
```

Signature according to man page:

```
int
     sendfile(int fd, int s, off_t offset, off_t *len, struct sf_hdtr *hdtr,
         int flags);
```

Fix doesn't seem too complicated, but I couldn't get rid of a compilation warning when casting size_t to off_t :-S

**Comment by bnoordhuis** [comment]
@saghul Can you try 75fd6de?

**Comment by saghul** [comment]
@bnoordhuis That fixes compilation :-) A related test fails though:

```
[%  80|+ 128|-   1]: fs_async_sendfile
`fs_async_sendfile` failed: exit code 134
Output from process `fs_async_sendfile`:
Assertion failed in test/test-fs.c on line 433: req->result == 65546
```

**Comment by bnoordhuis** [comment]
Fixed in 3977d1b.

**Comment by saghul** [comment]
Works like a charm :-)
